### PR TITLE
Add support for TimeLimitSec to Lexicographic

### PR DESCRIPTION
**Pull request by vcoppe**
Hi, I have added support for TimeLimitSec to the Lexicographic search algorithm.
- Given a fixed sequence of objectives, it returns the last solution found, even if not proved optimal for all objectives. The last found is very likely the best one found for this fixed sequence of objectives thanks to the additional constraints that are added after each iteration. (By the way, could we not warm-start each iteration with the solution previously found?)
- When solutions for all permutations of the objectives are asked, it simply collects all the solutions found by the above procedure for as many permutations as possible, whether optimal or not, and filters the dominated ones as before.
I hope that makes sense, I just started learning JuMP so it might not be perfect.

### CODE REVIEW

File: src/algorithms/Lexicographic.jl [hunk 1]
```diff
@@ -62,37 +65,53 @@ function MOI.set(alg::Lexicographic, ::LexicographicAllPermutations, val::Bool)
...
     sequence = 1:MOI.output_dimension(model.f)
     if !MOI.get(algorithm, LexicographicAllPermutations())
-        return _solve_in_sequence(algorithm, model, sequence)
+        return _solve_in_sequence(algorithm, model, sequence, start_time)
     end
     solutions = SolutionPoint[]
+    status = MOI.OPTIMAL
     for sequence in Combinatorics.permutations(sequence)
-        status, solution = _solve_in_sequence(algorithm, model, sequence)
+        status, solution = _solve_in_sequence(algorithm, model, sequence, start_time)
```
Comment (odow) [inline src/algorithms/Lexicographic.jl:76]:
```suggestion
        status, solution =
            _solve_in_sequence(algorithm, model, sequence, start_time)
```

**Comment by odow** [comment]
> By the way, could we not warm-start each iteration with the solution previously found?

We don't do this explicitly, but in-place solvers such as HiGHS/Gurobi will already have a solution in-memory that is automatically warm-started from.

> I just started learning JuMP so it might not be perfect.

A very nice first contribution!

Once CI runs I'll have a look at the coverage. We'd normally require a test, but testing time-limits can be quite tricky.

**Comment by vcoppe** [comment]
Great, tell me if can still help with anything

**Comment by odow** [comment]
The tests are failing, but I don't think it's your fault. I'm taking a look.

**Comment by odow** [comment]
Thanks!

**Pull request merged**